modules/home-manager/tmux.nix

```
@@ -18,12 +19,33 @@ let
 in
 {
   options.programs.tmux.catppuccin =
-    lib.ctp.mkCatppuccinOpt "tmux";
+    ctp.mkCatppuccinOpt "tmux"
+    // {
+      plugin = mkOption {
+        type = types.package;
+        default = plugin;
+        description = "Catppuccin tmux plugin";
+      };
+
+      extraConfig = mkOption {
+        type = types.lines;
+        description = "Additional configuration for the catppuccin plugin.";
+        default = "";
+        example = ''
+          set -g @catppuccin_status_modules_right "application session user host date_time"
+        '';
+      };
+    };
 
   config.programs.tmux.plugins = lib.mkIf enable [
     {
-      inherit plugin;
-      extraConfig = "set -g @catppuccin_flavour '${cfg.flavour}'";
+      inherit (cfg) plugin;
+      extraConfig = concatStrings [
+        ''
+          set -g @catppuccin_flavour '${cfg.flavour}'
+        ''
+        cfg.extraConfig
```

```suggestion
      extraConfig = lib.mkBefore ''
        set -g @catppuccin_flavour '${cfg.flavour}'
      '';
```
i don't think we really need to be re-exporting our parent option here either. `lib.mkBefore` can cover basically the same use case

---

The issue with `mkBefore` is that `programs.tmux.plugins` a list is. An I don't know any ~~good~~ way of modifying that `extraConfig`. Is there a way to modify it?

---

you're right, i must have had a brain fart there. this should be fine :)